ts/private/ts_validate_options.bzl

```diff
@@ -23,6 +24,13 @@ def _validate_options_impl(ctx):
     # We make it a .d.ts file so we can plumb it to the deps of the ts_project compile.
     marker = ctx.actions.declare_file("%s.optionsvalid.d.ts" % ctx.label.name)
 
+    # Provider validation
+    if not ctx.attr.allow_js:
+        for d in ctx.attr.deps:
+            if not d[JsInfo].declarations:
```

Comment:
I am not sure what would happen if there were an empty depset. would this condition still be true?

Reply:
An empty `depset` will evaluate to false. This is where the CI errors are failing atm.

https://github.com/aspect-build/rules_js/pull/563 fixes the CI errors failing due to npm packages missing `declarations`, but there's other failures we should discuss.

Reply:
gotcha. Afaik rules_js does not put declarations into `declarations`. It would have to do this at analysis time but there is no way to know what the package contains during analysis time.

Reply:
rules_js does it for `js_library` if there are dts, json, or directories with unknown content. I think that's an issue because if that changes (no more dts for example) then it will suddenly cause errors in `ts_project`s elsewhere. Maybe that's not an issue though?

rules_js will also do it for linked npm packages after https://github.com/aspect-build/rules_js/pull/563

I think that's all the places rules_js sets `JsInfo` so other then my concern with `js_library` it might be ok?